test/testdrive/kafka-avro-debezium-transaction.td

```diff
@@ -0,0 +1,365 @@
+{"status": "END", "id": "7", "event_count": {"long": 1}, "data_collections": {"array": [{"event_count": 1, "data_collection": "testdrive-data-${testdrive.seed}"}]}}
+
+> SELECT a, b FROM data_schema_inline
+a  b
+-----
+1  1
+2  3
+4  5
+8  9
+
+
+$ kafka-ingest format=avro topic=data schema=${schema} timestamp=1
+{"before": null, "after": {"row": {"a": 2, "b": 7}}, "source": {"file": "binlog3", "pos": 1, "row": 1, "snapshot": {"boolean": false}}, "transaction": {"total_order": null, "id": "7"}}
+
+> SELECT a, b FROM data_schema_inline
+a  b
+-----
+1  1
+2  3
+4  5
+8  9
+2  7
+
+$ sleep-is-probably-flaky-i-have-justified-my-need-with-a-comment duration=2s
+$ kafka-ingest format=avro topic=data schema=${schema} timestamp=1
+{"before": null, "after": {"row": {"a": 3, "b": 9}}, "source": {"file": "binlog4", "pos": 1, "row": 1, "snapshot": {"boolean": false}}, "transaction": {"total_order": null, "id": "9"}}
+
+> SELECT a, b FROM data_schema_inline
+a  b
+-----
+1  1
+2  3
+4  5
+8  9
+2  7
+
+$ kafka-ingest format=avro topic=data-txdata schema=${txschema} timestamp=2
+{"status": "BEGIN", "id": "9", "event_count": null, "data_collections": null}
+{"status": "END", "id": "9", "event_count": {"long": 1}, "data_collections": {"array": [{"event_count": 1, "data_collection": "testdrive-data-${testdrive.seed}"}]}}
+
+> SELECT a, b FROM data_schema_inline
+a  b
+-----
+1  1
+2  3
+4  5
+8  9
+2  7
+3  9
+
+#
+# Testing tx_metadata specification
+#
+$ kafka-create-topic topic=data-txdata-bad-schema
+$ kafka-create-topic topic=data-bad-schema
+
+> CREATE MATERIALIZED SOURCE data_txdata_bad_schema
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-txdata-bad-schema-${testdrive.seed}'
+  FORMAT AVRO USING SCHEMA '${txschema-bad-schema}'
+  ENVELOPE NONE
+
+
+$ kafka-ingest format=avro topic=data-txdata-bad-schema schema=${txschema-bad-schema} timestamp=4
+{"status": "BEGIN", "id": null, "event_count": null, "data_collections": null}
+{"status": "END", "id": {"string": "1"}, "event_count": {"long": 3}, "data_collections": {"array": [{"event_count": 3, "data_collection": "testdrive-data-${testdrive.seed}"}]}}
+
+! CREATE MATERIALIZED SOURCE data_schema_inline_with_bad_schema_tx_metadata
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-bad-schema-${testdrive.seed}'
+  WITH(tx_metadata = data_txdata_bad_schema, tx_metadata_collection_name = 'testdrive-data-bad-schema-${testdrive.seed}')
+  FORMAT AVRO USING SCHEMA '${schema}'
+  ENVELOPE DEBEZIUM
+contains:'id' column must be of type non-nullable string
+
+! CREATE MATERIALIZED SOURCE data_schema_inline_with_sink
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-${testdrive.seed}'
+  WITH(tx_metadata = data_sink, tx_metadata_collection_name = 'testdrive-data-${testdrive.seed}')
+  FORMAT AVRO USING SCHEMA '${schema}'
+  ENVELOPE DEBEZIUM
+contains:tx_metadata catalog item doesn't have a desc
```

Comment:
Hm, should we give a more descriptive error like "tx_metadata must refer to a source, table, or view"?